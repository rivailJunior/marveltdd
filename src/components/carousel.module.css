.showCarousel {
    display: block;
}

.hideCarousel {
    display: none;
}