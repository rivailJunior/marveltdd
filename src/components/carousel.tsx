import { useState } from "react";
import { Carousel } from "react-bootstrap";
import styles from "./carousel.module.css";

type iCarouselProps = {
    data: [any];
    type: string;
};
export function CarouselInfor({ data, type }: iCarouselProps): JSX.Element {
    const [index, setIndex] = useState(0);

    const handleSelect = (selectedIndex: number, e: any) => {
        setIndex(selectedIndex);
    };

    const showErr = !data.length ? styles.showCarousel : styles.hideCarousel;
    const showCarousel = data.length
        ? styles.showCarousel
        : styles.hideCarousel;
    return (
        <>
            <div className={showErr}>Ops. Nenhuma {type} encontrado</div>
            <Carousel
                className={showCarousel}
                activeIndex={index}
                onSelect={handleSelect}
            >
                {data.map((item, index) => {
                    <Carousel.Item key={index}>
                        <img
                            className="d-block w-100"
                            src="holder.js/800x400?text=First slide&bg=373940"
                            alt="First slide"
                        />
                        <Carousel.Caption>
                            <h3>First slide label</h3>
                            <p>
                                Nulla vitae elit libero, a pharetra augue mollis
                                interdum.
                            </p>
                        </Carousel.Caption>
                    </Carousel.Item>;
                })}
            </Carousel>
        </>
    );
}
