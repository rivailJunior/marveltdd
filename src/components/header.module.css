.marvelHeader {
    /* padding: 10%; */
    top: 0px;
    left: 0px;
    width: 100%;
    height: 64px;
    background: #FFFFFF 0% 0% no-repeat padding-box;
    opacity: 1;
}

.labelName {
    text-align: left;
    font: normal normal bold 14px/20px PT Sans Caption;
    letter-spacing: 0px;
    color: #555555;
}

.informationProps {
    height: 100%;
    width: 100%;
    background-color: lightcoral;
}