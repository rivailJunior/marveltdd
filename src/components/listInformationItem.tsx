import styles from "./marvelList.module.css";

type iListProps = {
    list: [any] | any;
    tagName: string;
    information?: string;
};
export const ListInformationItem = ({
    list,
    tagName,
    information,
}: iListProps): JSX.Element => {
    return list?.length > 0 ? (
        <ul className={styles.list}>
            {list.slice(0, 3).map((item: any, index: number) => {
                return (
                    <li data-testid="informationLi" key={index}>
                        {item[tagName]}
                    </li>
                );
            })}
        </ul>
    ) : (
        <div className={styles.defaultLabel}>
            Ops. {information} n√£o encontrados!
        </div>
    );
};
