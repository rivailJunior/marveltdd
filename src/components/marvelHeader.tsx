import { Navbar } from "react-bootstrap";
import styles from "./header.module.css";

type iHeaderProps = {
    leftLabel?: string;
    userName?: string;
    descriptionRight?: string;
};

const MarvelHeader = ({
    leftLabel,
    userName,
    descriptionRight,
}: iHeaderProps): JSX.Element => {
    const NoPropsDivInfor = () => {
        return (
            <div
                className={styles.informationProps}
                data-testid="informationProps"
            >
                ops!
            </div>
        );
    };

    return (
        <Navbar
            expand="lg"
            variant="light"
            bg="light"
            className={styles.marvelHeader}
        >
            <Navbar.Brand href="#home">
                {!leftLabel?.length ? NoPropsDivInfor() : leftLabel}
            </Navbar.Brand>
            <Navbar.Toggle />

            <Navbar.Collapse className="justify-content-end">
                <Navbar.Text className={styles.labelName}>
                    {!userName?.length ? NoPropsDivInfor() : userName}
                    {!descriptionRight?.length ? (
                        NoPropsDivInfor()
                    ) : (
                        <a href="#login">{descriptionRight}</a>
                    )}
                </Navbar.Text>
            </Navbar.Collapse>
        </Navbar>
    );
};

export { MarvelHeader };
