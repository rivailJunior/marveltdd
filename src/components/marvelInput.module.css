.showClear {
    display: block;
}

.hideClear {
    display: none !important;
}