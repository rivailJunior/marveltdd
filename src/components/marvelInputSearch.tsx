import { useState, useEffect } from "react";
import { InputGroup, FormControl } from "react-bootstrap";
import styles from "./marvelInput.module.css";

type iInputProps = {
    handleChangeName: Function;
};
const MarvelInputSearch = ({ handleChangeName }: iInputProps) => {
    const [name, setName] = useState("");
    useEffect(() => {
        handleChangeName(name);
    }, [name, handleChangeName]);
    const showOrHideClear =
        name.length > 0 ? styles.showClear : styles.hideClear;
    return (
        <InputGroup className="mb-3">
            <FormControl
                value={name}
                onChange={(evt) => setName(evt.target.value)}
                data-testid="inputSearch"
                placeholder="Search"
                aria-label="Busque personagens"
            />
            <InputGroup.Append data-testid="btnSearch">
                <InputGroup.Text>OK</InputGroup.Text>
            </InputGroup.Append>
            <InputGroup.Append
                className={showOrHideClear}
                data-testid="btnClear"
                onClick={() => setName("")}
            >
                <InputGroup.Text>X</InputGroup.Text>
            </InputGroup.Append>
        </InputGroup>
    );
};

export { MarvelInputSearch };
