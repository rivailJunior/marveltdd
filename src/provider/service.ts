import { Character, CharacterDataContainer } from './../characterTypes/characters';
import axios, { AxiosResponse } from 'axios'
const apikey = process.env.REACT_APP_PUBKEY
// const hash = process.env.REACT_APP_HASH
// const ts = 1;

const api = axios.create({
    baseURL: 'https://gateway.marvel.com/v1/public',
});

const getService = async (url: string, options: object): Promise<AxiosResponse> => {
    return await api.get(url, { params: { apikey, ...options } })
}

const getCharacters = async (total: number): Promise<CharacterDataContainer> => {
    const result = await getService('/characters', { offset: 0, limit: total })
    return result.data?.data
}

const getParticipation = async (id: number, type: string): Promise<any> => {
    const result = await getService(`/characters/${id}/${type}`, {})
    return result.data?.data
}

const getCharacter = async (id: number): Promise<Character> => {
    const result = await getService(`/characters/${id}`, {})
    return result.data?.data?.results
}

export { getCharacters, getCharacter, getParticipation as getParticipations };