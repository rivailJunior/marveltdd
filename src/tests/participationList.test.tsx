import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { getCharacter } from "../provider/service";
import { ListInformationItem } from "../components/listInformationItem";

afterEach(cleanup);

describe("`<ParticipationListItem>`", () => {
    test("Render correctly", () => {
        const container = renderer.create(
            <ListInformationItem
                list={[]}
                tagName="name"
                information="Eventos"
            />
        );
        container.toJSON();
        expect(container).toMatchSnapshot();
    });

    test("Render with no item", () => {
        const { getByText } = render(
            <ListInformationItem
                list={[]}
                tagName="name"
                information="Eventos"
            />
        );
        expect(getByText(/Ops. Eventos nÃ£o encontrados!/i)).toBeInTheDocument();
    });

    test("Render max 3 itens on list", async () => {
        const response = await getCharacter(1009148);
        const { findAllByTestId } = render(
            <ListInformationItem
                list={response[0]?.events?.items}
                tagName="name"
                information="Eventos"
            />
        );
        const list = await findAllByTestId("informationLi");
        expect(list.length).toBeLessThan(4);
    });
});
