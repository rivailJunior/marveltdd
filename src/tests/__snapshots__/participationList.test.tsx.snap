// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<ParticipationListItem>\` Render correctly 1`] = `
<div
  className="defaultLabel"
>
  Ops. 
  Eventos
   n√£o encontrados!
</div>
`;
